msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/AbstractMsalCustomBrokerInstallationTest.java

```diff
@@ -0,0 +1,95 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.broker;
+
+import com.microsoft.identity.client.msal.automationapp.AbstractMsalUiTest;
+import com.microsoft.identity.client.ui.automation.ICustomBrokerInstallationTest;
+import com.microsoft.identity.client.ui.automation.broker.BrokerCompanyPortal;
+import com.microsoft.identity.client.ui.automation.broker.BrokerHost;
+import com.microsoft.identity.client.ui.automation.broker.BrokerLTW;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+
+/**
+ * Msal UI Testing class that allows devs to easily install old/legacy and updated versions of brokers
+ * quickly to test complex update scenarios.
+ */
+public abstract class AbstractMsalCustomBrokerInstallationTest extends AbstractMsalUiTest implements ICustomBrokerInstallationTest {
+
+    // Fields to access installed brokers
+    public BrokerHost mBrokerHost;
+    public BrokerMicrosoftAuthenticator mAuthenticator;
+    public BrokerCompanyPortal mCompanyPortal;
+    public BrokerLTW mLtw;
+
+    @Override
+    public void installOldBrokerHost() {
+        mBrokerHost = new BrokerHost(BrokerHost.OLD_BROKER_HOST_APK,
+                BrokerHost.BROKER_HOST_APK);
```

Comment:
do we still need to pass the new broker host apk name here? Or can we move this parameter to be passed in installBrokerHost method?

Reply:
So, I took a look.

![image](https://user-images.githubusercontent.com/88730756/230669143-9f0e60c1-3cd7-4ef5-afb9-2a0df233d340.png)

This is if we pass a singular apk name. Looks like it's used for both the initial install and the update file name, so I would think that the update call would be broken here. I'm going to leave it as is for now.